samples: usb: hid-mouse: multiple buttons

masz-nordic opened the pull request:
Added right button, X and Y axis movement to HID mouse example.
Enabled only if board has enough buttons.
Added LED toggling notification when report is sent.

Signed-off-by: Marcin Szymczyk <Marcin.Szymczyk@nordicsemi.no>

finikorg commented on samples/subsys/usb/hid-mouse/src/main.c:
```diff
@@ -92,75 +149,178 @@ static const u8_t hid_report_desc[] = {
+}
+#endif
+
+int configure_callback(struct device *gpio, u32_t pin, int flags,
```
Is this a callback?

masz-nordic replied:
No, it configures GPIOs and adds callbacks to button pins. I see now this might not be the best name. `callbacks_configure()` maybe?

finikorg replied:
Yes, this sound better, at first it sounds like a callback

finikorg commented on samples/subsys/usb/hid-mouse/src/main.c:
```diff
@@ -92,75 +149,178 @@ static const u8_t hid_report_desc[] = {
+
+#ifdef SW1_GPIO_PIN
+static void right_button(struct device *gpio, struct gpio_callback *cb,
+		  u32_t pins)
```
is this line aligned?

finikorg commented:
You need to add information to readme also

qianfan-Zhao commented on samples/subsys/usb/hid-mouse/src/main.c:
```diff
@@ -92,75 +149,178 @@ static const u8_t hid_report_desc[] = {
 		/* END_COLLECTION				C0    */
 };
 
-static int get_report_cb(struct usb_setup_packet *setup, s32_t *len,
```
Based on our discussion on Slack, this is a bug. I prefer move this fix to another PR.

qianfan-Zhao commented on samples/subsys/usb/hid-mouse/src/main.c:
```diff
@@ -92,75 +149,178 @@ static const u8_t hid_report_desc[] = {
 		/* END_COLLECTION				C0    */
 };
 
-static int get_report_cb(struct usb_setup_packet *setup, s32_t *len,
```
Based on our discussion on Slack, this is a bug. I prefer move this fix to another PR

masz-nordic replied:
Removed from this PR, I will make a new one specifically for this bugfix.

qianfan-Zhao commented on samples/subsys/usb/hid-mouse/src/main.c:
```diff
@@ -18,22 +22,75 @@
+#define PIN1	SW1_GPIO_PIN
+#endif
+
+#ifdef SW1_GPIO_NAME
```
Actually, the first version I wrote, used this marco to detect if a PIN was defined or not. But in that times Zephyr doesn't support write BUTTON node in dts, somethings like SW0_GPIO_CONTROLLER are defined in board.h

But Zephyr support adding BUTTON node in dts now, so this check is useless. I think somethings like https://github.com/zephyrproject-rtos/zephyr/blob/93fc23bc7e458f9155c59980a86226a00b69aa9a/samples/basic/blinky/src/main.c#L11 is enough.

@galak Could you review this sections?

jfischer-phytec-iot replied:
It seems that we can simplify it and use only  SW*_GPIO_CONTROLLER, SW*_GPIO_FLAGS, SW*_GPIO_PIN now

masz-nordic replied:
Removed SW*_GPIO_NAME ifdefs.

jfischer-phytec-iot commented on samples/subsys/usb/hid-mouse/src/main.c:
```diff
@@ -105,62 +162,174 @@ static struct hid_ops ops = {
 
-static int def_val;
-static volatile int status;
+static u32_t def_val[4];
```
It may be correct, but not nice to change the type of the variable in this commit.

masz-nordic replied:
Should I make a separate one?

jfischer-phytec-iot replied:
I think yes, would be more accurate :-)

masz-nordic replied:
Done :)

masz-nordic committed:
samples: usb: hid-mouse: type change

Change def_val, cur_val and status type from int to u32_t, u32_t
and u8_t.

Signed-off-by: Marcin Szymczyk <Marcin.Szymczyk@nordicsemi.no>

masz-nordic committed:
samples: usb: hid-mouse: multiple buttons

Added right button, X and Y axis movement to HID mouse example.
Enabled only if board has enough buttons.
Added LED toggling notification when report is sent.

Signed-off-by: Marcin Szymczyk <Marcin.Szymczyk@nordicsemi.no>

qianfan-Zhao commented on samples/subsys/usb/hid-mouse/src/main.c:
```diff
@@ -105,63 +147,174 @@ static struct hid_ops ops = {
+	if (def_val[2] != cur_val) {
+		state += 10;
+	} else {
+
```
How about remove this unused else?

masz-nordic replied:
Removed, thanks.

masz-nordic commented:
@dbkinder @finikorg please recheck

dbkinder reviewed (approved):
LGTM, thanks.
